src/app/public/modules/viewkeeper/viewkeeper.service.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  Injectable,
+  Optional
+} from '@angular/core';
+
+import {
+  SkyViewkeeper
+} from './viewkeeper';
+
+import {
+  SkyViewkeeperHostOptions
+} from './viewkeeper-host-options';
+
+import {
+  SkyViewkeeperOptions
+} from './viewkeeper-options';
+
+@Injectable()
+export class SkyViewkeeperService {
+
+  constructor(@Optional() private hostOptions?: SkyViewkeeperHostOptions) { }
+
+  public create(options: SkyViewkeeperOptions): SkyViewkeeper {
```

Comment:
Just confirming: Since this is an Angular service, it might be odd from a documentation standpoint to provide options that include native HTML values (such as `el: HTMLElement`). I know you mentioned keeping this feature as "vanilla" as possible in case we converted it to a strictly JavaScript library, but I wanted to bring this up to make sure we were comfortable using the same options object for the Angular stuff as this isn't customary with our other components' services.

Reply:
I expect viewkeeper to be an internal implementation for SKY UX components for the foreseeable future, which is why I'm not too concerned with providing a slick Angular-ized API for public consumption. Also, if you use viewkeeper with its main use case, i.e. through the directive, then consumers will be specifying CSS selectors in the template as opposed to having a handle to an `HTMLElement` in the component.